### DISCUSSION TITLE
Serial monitor showing "Connect to AC101 codec Failed!". [solved]

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#37] [Q&A]
The 'esp32 _midi_sampler' program was successfully uploaded to ESP32 audio-kit. But when I open the serial monitor it shows "Connect to AC101 codec Failed!".
Please help me out.

![IMG20210725135352](https://user-images.githubusercontent.com/53053463/126892734-7ae3248e-b9aa-4259-a169-9d32ffdf189f.jpg)

**Comment by marcel-licence** [comment]
Is it the first time you are trying to use it? Do you have a picture of the Audio Kit?
I can recommend running I2C scanner to ensure the AC101 can be connected `Wire.begin(33, 32)`;.
If that fails you can try a different pin setting: `Wire.begin(18,23);`

**Reply by souravlal** [reply]
![IMG20210725150556](https://user-images.githubusercontent.com/53053463/126894749-9136c27c-26e1-405e-92cc-045043b0fda4.jpg)
![IMG20210725150622](https://user-images.githubusercontent.com/53053463/126894765-2a84e6f3-29be-45d0-a32d-76adf68d1447.jpg)

@marcel-licence I2C Scanner or I2S scanner?

**Reply by marcel-licence** [reply]
I2C scanner. The ESP32 A1S has a black coated antenna on your picture. I guess that this one doesn't have the AC101 codec on board. I guess it will use the ES8388 instead. You can use the scanner to try to confirm which codec is placed. Unfortunately I don't have any further information about the ES8388. You may find information in other forums.

**Reply by souravlal** [reply]
![IMG20210725161131](https://user-images.githubusercontent.com/53053463/126896299-6d46760a-8d46-42cb-84c9-b2122092bb49.jpg)

I have run the i2c scanner with "Wire.begin(18,23)"  it shows one device connected.

**Reply by marcel-licence** [reply]
Okay I think the code will require some modification to get the codec working. I2C is used to configure the codec. I2S will be used after configuration just for audio data. I never worked with the ES8388. It is also just a guess that the ES8388 could be placed here. If you manage to replace the AC101 config code with another one which can set up the ES8388 to 16 bit 44100 Hz then the rest of the project could work.

**Reply by souravlal** [reply]
let me try

**Reply by marcel-licence** [reply]
I've uploaded an update today which does support the ES8388

**Reply by souravlal** [reply]
Oh, Thanks.
I have already bought the one with AC101.

@marcel-licence Does the ESP32 Audio-kit require a restart every time we need to load a new sample?This is the case with me.

**Reply by marcel-licence** [reply]
A restart shouldn't be required. You should ensure that you have enough memory left (~4MB is available for all samples).

**Reply by marcel-licence** [reply]
https://youtu.be/8UB3fYPjqSk a video is also online using the ES8388

**Reply by souravlal** [reply]
👍
